File: js/best_time_121.js
var best = (nums) => {
  let i = 0,
    j = 1,
    maxP = 0;

  while (i < nums.length && j < nums.length) {
    let diff = nums[j] - nums[i];

    if (diff < 0) {
      i++;
      j = i + 1;
    } else {
      maxP = Math.max(maxP, diff);
      j++;
    }
  }

  return maxP;
};

const resp = best([7, 1, 5, 3, 6, 4]);
console.log("resp", resp);

File: dart/best_time_121.dart
import 'dart:math' as Math;

void main() {
  final resp = maxArea([7, 1, 5, 3, 6, 4]);
  print(resp);
}

int maxArea(List<int> nums) {
  int l = 0, r = 1;
  int maxP = 0;

  while (l < nums.length && r < nums.length) {
    int diff = nums[r] - nums[l];

    if (diff < 0) {
      l++;
      r = l + 1;
    } else {
      maxP = Math.max(maxP, diff);
      r++;
    }
  }

  return maxP;
}
